Gems/Atom/Feature/Common/Code/Source/CoreLights/CapsuleLightFeatureProcessor.cpp

```diff
@@ -107,8 +107,7 @@ namespace AZ
 
             if (m_deviceBufferNeedsUpdate)
             {
-                [[maybe_unused]] bool success = m_lightBufferHandler.UpdateBuffer(m_capsuleLightData.GetDataVector());
-                AZ_Error(FeatureProcessorName, success, "Unable to update buffer during Simulate().");
+                m_lightBufferHandler.UpdateBuffer(m_capsuleLightData.GetDataVector());
                 m_deviceBufferNeedsUpdate = false;
```

Comment:
Would this be better as
`m_deviceBufferNeedsUpdate = !m_lightBufferHandler.UpdateBuffer(m_capsuleLightData.GetDataVector());`

Reply:
This is repeated over several files in this PR

Reply:
Making the change you recommended would mean that it will keep trying to update the buffer until it succeeds. In theory it sounds good to me but I am hesitant to make this change as we use the current pattern everywhere in a bunch of other FPs. I would prefer the change be done separately in a separate pass.